pkg/controller/jobframework/reconciler.go

```diff
@@ -607,8 +613,39 @@ func (r *JobReconciler) ensurePrebuiltWorkloadInSync(ctx context.Context, wl *ku
 	return true, nil
 }
 
+// get the expected podsets during the job execution, returns nil if the workload has no reservation or
+// the admission dose not fit.
+func expectedRunningPodSets(ctx context.Context, c client.Client, wl *kueue.Workload) []kueue.PodSet {
+	if !workload.HasQuotaReservation(wl) {
+		return nil
+	}
+	info, err := getPodSetsInfoFromStatus(ctx, c, wl)
+	if err != nil {
+		return nil
+	}
+	infoMap := slices.ToRefMap(info, func(psi *podset.PodSetInfo) string { return psi.Name })
+	runningPodSets := wl.Spec.DeepCopy().PodSets
+	canBePartiallyAdmitted := workload.CanBePartiallyAdmitted(wl)
+	for i := range runningPodSets {
+		ps := &runningPodSets[i]
+		psi, found := infoMap[ps.Name]
+		if !found {
+			return nil
+		}
+		err := podset.Merge(&ps.Template.ObjectMeta, &ps.Template.Spec, *psi)
+		if err != nil {
+			return nil
+		}
+		if canBePartiallyAdmitted && ps.MinCount != nil {
+			// update the expected running count
+			ps.Count = psi.Count
+		}
```

Comment:
can this just be:
```
psi.Count = psi.Count
```
?

If the value is set, admission already determined if partial admission was possible.